plugins/keyboard-navigation/src/navigation_controller.js

```diff
@@ -0,0 +1,208 @@
+   * @param {!Blockly.ShortcutRegistry.KeyboardShortcut} shortcut The shortcut
+   *     to be handled.
+   * @return {boolean} True if the field handled the shortcut,
+   *     false otherwise.
+   * @this {Blockly.FieldDropdown}
+   * @protected
+   */
+  fieldDropdownHandler(shortcut) {
+    if (this.menu_) {
+      switch (shortcut.name) {
+        case Constants.SHORTCUT_NAMES.PREVIOUS:
+          this.menu_.highlightPrevious();
+          return true;
+        case Constants.SHORTCUT_NAMES.NEXT:
+          this.menu_.highlightNext();
+          return true;
+        default:
+          return false;
+      }
+    }
+    return Blockly.FieldDropdown.superClass_.onShortcut.call(this, shortcut);
+  }
+
+  /**
+   * Handles the given keyboard shortcut.
+   * This is only triggered when keyboard accessibility mode is enabled.
+   * @param {!Blockly.ShortcutRegistry.KeyboardShortcut} shortcut The shortcut
+   *     to be handled.
+   * @return {boolean} True if the toolbox handled the shortcut,
+   *     false otherwise.
+   * @this {Blockly.Toolbox}
+   * @protected
+   */
+  toolboxHandler(shortcut) {
+    if (!this.selectedItem_) {
+      return false;
+    }
+    switch (shortcut.name) {
+      case Constants.SHORTCUT_NAMES.PREVIOUS:
+        return this.selectPrevious();
+      case Constants.SHORTCUT_NAMES.OUT:
+        return this.selectParent();
+      case Constants.SHORTCUT_NAMES.NEXT:
+        return this.selectNext();
+      case Constants.SHORTCUT_NAMES.IN:
+        return this.selectChild();
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Adds all necessary event listeners and markers to a workspace for keyboard
+   * navigation to work. This must be called for keyboard navigation to work
+   * on a workspace.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to add keyboard
+   *     navigation to.
+   * @public
+   */
+  addWorkspace(workspace) {
+    this.navigation.addWorkspace(workspace);
+  }
+
+  /**
+   * Removes all necessary event listeners and markers to a workspace for
+   * keyboard navigation to work.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to remove keyboard
+   *     navigation from.
+   * @public
+   */
+  removeWorkspace(workspace) {
+    this.navigation.removeWorkspace(workspace);
+  }
+
+  /**
+   * Turns on keyboard navigation.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to turn on keyboard
+   *     navigation for.
+   * @public
+   */
+  enable(workspace) {
+    this.navigation.enableKeyboardAccessibility(workspace);
+  }
+
+  /**
+   * Turns off keyboard navigation.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to turn off keyboard
+   *     navigation on.
+   * @public
+   */
+  disable(workspace) {
+    this.navigation.disableKeyboardAccessibility(workspace);
+  }
+
+
+  /**
+   * Removes all the keyboard navigation shortcuts.
+   * @public
+   */
+  dispose() {
```

Comment:
This removes the shortcuts but leaves the handlers in place, right?

Reply:
Yup, good point. Added a `removeShortcutHanlders` method.